ibis/expr/types/temporal.py

```diff
@@ -19,9 +48,215 @@ class TemporalColumn(AnyColumn, TemporalValue):
     pass  # noqa: E701,E302
 
 
+class DateComponentValue(TemporalValue):
+    """Temporal expressions that have a date component.
+
+    Currently this includes
+    [`TimestampValue`][ibis.expr.types.temporal.TimestampValue]s and
+    [`DateValue`][ibis.expr.types.temporal.DateValue]s
+    """
+
+    def epoch_seconds(self) -> ir.IntegerValue:
+        """Extract UNIX epoch in seconds."""
+        import ibis.expr.operations as ops
+
+        return ops.ExtractEpochSeconds(self).to_expr()
+
+    def year(self) -> ir.IntegerValue:
+        """Extract the year component."""
+        import ibis.expr.operations as ops
+
+        return ops.ExtractYear(self).to_expr().name("year")
+
+    def month(self) -> ir.IntegerValue:
+        """Extract the month component."""
+        import ibis.expr.operations as ops
+
+        return ops.ExtractMonth(self).to_expr().name("month")
+
+    def day(self) -> ir.IntegerValue:
+        """Extract the day component."""
+        import ibis.expr.operations as ops
+
+        return ops.ExtractDay(self).to_expr().name("day")
+
+    @property
+    def day_of_week(self) -> DayOfWeek:
+        """Return a namespace containing methods for extracting day of week information.
+
+        Returns
+        -------
+        DayOfWeek
+            An namespace expression containing methods to use to extract
+            information.
+        """  # noqa: E501
+
+        import ibis.expr.operations as ops
+
+        return ops.DayOfWeekNode(self).to_expr()
+
+    def day_of_year(self) -> ir.IntegerValue:
+        """Extract the day of the year component."""
+        import ibis.expr.operations as ops
+
+        return ops.ExtractDayOfYear(self).to_expr()
+
+    def quarter(self) -> ir.IntegerValue:
+        """Extract the quarter component."""
+        import ibis.expr.operations as ops
+
+        return ops.ExtractQuarter(self).to_expr()
+
+    def week_of_year(self) -> ir.IntegerValue:
+        """Extract the week of the year component."""
+        import ibis.expr.operations as ops
+
+        return ops.ExtractWeekOfYear(self).to_expr()
+
+
+class TimeComponentValue(TemporalValue):
```

Comment:
Is it useful on its own or is it actually used as a mixin? In the latter case I'd prefer a protected mixin instead: `_TimeComponentMixin`